internal/pkg/otel/extension/elasticdiagnostics/README.md

```diff
@@ -0,0 +1,62 @@
+# elasticdiagnosticsextension
+
+`elasticdiagnosticsextension` is an internal package for peforming diagnostics and is used in conjunction with EDOT.
```

Comment:
Mention that it's designed to return diagnostics compatible with the control protocols diagnostics action result type https://github.com/elastic/elastic-agent-client/blob/888026ef85e1c9190fe76eb158cf21d9c9c02920/elastic-agent-client.proto#L424-L437